File: CSharpCodeBase/components/builder/buildercomponent.lua
local Class = require 'utils.hump.class'
local Component = require 'components.component'

local BuilderComponent = Class({__includes=Component, init=function(self, entity)
      Component.init(self, entity)
      self.entity.builder = self
      self.components = self.entity.config:Get('component')
      
      if self.components then
        for key, value in pairs(self.components) do
          self.entity:AddComponent(require(value))
        end
      end
end})

return BuilderComponent

File: CSharpCodeBase/components/builder/buildercomponent.cs
using System;
 
 using System.Collections.Generic;
 
 using System.Linq;
 
 using System.Text;
 
 using UnityEngine;
 
 using System.Collections;
 
 using MainGame.core;
 
 using UnityEngine.EventSystems;
 
 namespace MainGame{
 public class buildercomponent :component{
 public void init(self, entity);
       Component.init(self, entity);
       this.entity.builder = self;
       this.components = this.entity.config:Get("component");
       
       if(this.components  ){ 
         for(key, value in pairs(this.components) ){
           this.entity:AddComponent(require(value));
         }
       }
 }})
 }}